src/main/java/tempest/GUI/AddSessionPage.java

```diff
@@ -0,0 +1,184 @@
+    public void actionPerformed(ActionEvent e){
+        Object source = e.getSource();
+
+        if(source == enterButton){
+            handleAddingSession();
+        }
+    }
+
+    /**
+     * Handles the user trying to enter a session
+     */
+    private void handleAddingSession() {
+        if (moduleDropDown.getItemCount() < 1) { // No modules created, so can't add a session
+            return;
+        }
+
+        String moduleName = Objects.requireNonNull(moduleDropDown.getSelectedItem()).toString();
+
+        Date date = new Date();
+
+        String hours = hoursInput.getText();
+        String minutes = minutesInput.getText();
+
+        int hoursInt = 0;
+        int minutesInt = 0;
+
+        hours = hours.strip();
+        minutes = minutes.strip();
+
+        if (hours.equals("") && minutes.equals("")) { // Empty hours and minutes
+            System.out.println("A session needs to be longer >= 1 minute");
+            return;
+        }
+        else if (hours.equals("")) { // Only minutes have been entered
+            try {
+                minutesInt = Integer.parseInt(minutes);
+
+                if (minutesInt <= 0) {
+                    throw new NumberFormatException(); // Negative minutes
+                }
+            } catch (NumberFormatException e) {
+                System.out.println("Invalid minutes entered");
+                return;
+            }
+        }
+        else if (minutes.equals("")) { // Only hours have been entered
+            try {
+                hoursInt = Integer.parseInt(hours);
+
+                if (hoursInt <= 0) {
+                    throw new NumberFormatException(); // Negative hours
+                }
+            } catch (NumberFormatException e) {
+                System.out.println("Invalid hours entered");
+                return;
+            }
+        }
+        else { // Hours and minutes have been entered
+            try {
+                minutesInt = Integer.parseInt(minutes);
+                hoursInt = Integer.parseInt(hours);
+
+                if (hoursInt <= 0 || minutesInt <= 0) {
+                    throw new NumberFormatException();
+                }
+            } catch (NumberFormatException e) {
+                System.out.println("Invalid time entered");
+                return;
+            }
+        }
+
+        Duration time = Duration.ofMinutes(hoursInt * 60L + minutesInt);
+
+        boolean foundName = false;
+
+        for (Module m : state.getModules()) {
+            if (moduleName.equals(m.getName())) {
+                m.addSession(date, time);
+                foundName = true;
+                break;
+            }
+        }
+
+        if (!foundName) {
+            System.out.println("Unable to find this module");
+            return;
+        }
+
+        System.out.println("Study session successfully added");
+
+        hoursInput.setText(""); // Clearing inputted hours
+        minutesInput.setText(""); // Clearing inputted mins
```

Comment:
Do you think it would be appropriate to refactor this to a new method and add a button for clearing the input fields as well?

Reply:
Done and done